python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr_serverless.py

```diff
@@ -0,0 +1,289 @@
+import sys
+import time
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+import boto3
+import dagster._check as check
+from dagster import PipesClient
+from dagster._annotations import public
+from dagster._core.definitions.resource_annotation import TreatAsResourceParam
+from dagster._core.errors import DagsterExecutionInterruptedError
+from dagster._core.execution.context.compute import OpExecutionContext
+from dagster._core.pipes.client import (
+    PipesClientCompletedInvocation,
+    PipesContextInjector,
+    PipesMessageReader,
+)
+from dagster._core.pipes.context import PipesSession
+from dagster._core.pipes.utils import PipesEnvContextInjector, open_pipes_session
+
+from dagster_aws.pipes.message_readers import PipesCloudWatchMessageReader
+
+if TYPE_CHECKING:
+    from mypy_boto3_emr_serverless.client import EMRServerlessClient
+    from mypy_boto3_emr_serverless.type_defs import (
+        GetJobRunResponseTypeDef,
+        JobRunStateType,
+        StartJobRunRequestRequestTypeDef,
+        StartJobRunResponseTypeDef,
+    )
+
+AWS_SERVICE_NAME = "EMR Serverless"
+AWS_SERVICE_NAME_LOWER = AWS_SERVICE_NAME.replace(" ", "").lower()
+AWS_SERVICE_NAME_LOWER_KEBAB = AWS_SERVICE_NAME.lower().replace(" ", "-").replace("_", "-")
+BOTO3_START_METHOD = "run_job_flow"
+
+
+class PipesEMRServerlessClient(
+    PipesClient,
+    TreatAsResourceParam,
+):
+    f"""A pipes client for running workloads on AWS {AWS_SERVICE_NAME}.
+
+    Args:
+        client (Optional[boto3.client]): The boto3 {AWS_SERVICE_NAME} client used to interact with AWS {AWS_SERVICE_NAME}.
+        context_injector (Optional[PipesContextInjector]): A context injector to use to inject
+            context into AWS {AWS_SERVICE_NAME} workload. Defaults to :py:class:`PipesEnvContextInjector`.
+        message_reader (Optional[PipesMessageReader]): A message reader to use to read messages
+            from the {AWS_SERVICE_NAME} workload. Defaults to :py:class:`PipesCloudWatchMessageReader`.
+        forward_termination (bool): Whether to cancel the {AWS_SERVICE_NAME} workload if the Dagster process receives a termination signal.
```

Comment:
Not sure doing this variable Interpolation. I'd rather just see the docstring as it will appear in our docs.

Reply:
This is a leftover from the `BaseAWSPipesClient` PR. 
I wanted to generalize docstrings over the AWS service. 
I'll fix this here.

Reply:
If you want to enforce consistency recommend doing it via a unit test.

Reply:
Not exactly, just trying to standardize boilerplate such as this one. 
Basically, a lot of AWS pipes clients will share these exact docstrings with the only difference being the service name. 
Same with log messages. 
But let's keep it simple until we refactor different classes to extend `BaseAWSPipesClient`.